tenpy/algorithms/mps_sweeps.py

```diff
@@ -240,38 +253,44 @@ def get_sweep_schedule(self):
 
         Returns
         -------
-        schedule_i0 : list
-            List of indices of 'active sites'.
-        update_LP_RP : list
-            List of bools, which indicate whether to update the `LP` and `RP`.
+        schedule : iterable of (int, bool, (bool, bool))
+            Schedule for the sweep. Each entry is ``(i0, move_right, (update_LP, update_RP))``,
+            where `i0` is the leftmost of the ``self.EffectiveH.length` sites to be updated in
+            :meth:`update_local`, `move_right` indicates whether the next `i0` in the schedule is
+            rigth (`True`) of the current one, and `update_LP`, `update_RP` indicate
+            whether it is necessary to update the `LP` and `RP`.
+            The latter are chosen such that the environment is growing for infinite systems, but
+            we only keep the minimal number of environment tensors in memory.
         """
         L = self.psi.L
         if self.finite:
-            schedule_i0 = list(range(0, L - 1)) + list(range(L - 3, 0, -1))
-            update_LP_RP = [[True, False]] * (L - 2) + [[False, True]] * (L - 2)
+            l = self.EffectiveH.length
+            assert L >= l
+            i0s = list(range(0, L - l)) + list(range(L - l, 0, -1))
```

Comment:
Is this correct? Comparing to the old case, (which should be equal to `l=2`, right?), these ranges are now different.

Reply:
@jhauschild Could you confirm or debunk this?

Reply:
The infinite case is still the same.

In the finite case, it was `list(range(0, L - 1)) + list(range(L - 3, 0, -1) == list(range(0, L-2)) + list(range(L-2, 0, -1))` before.
Now it is `list(range(0, L-n)) + list(range(L-n, 0, -1))`, so this is still the same as well.

Reply:
Oh snap! Of course `list(range(0, L - 1)) + list(range(L - 3, 0, -1) == list(range(0, L-2)) + list(range(L-2, 0, -1))`. 

Apologies.